### PR TITLE
UI Team: refactoring workspace logic

### PR DESCRIPTION
**Pull request by daysleeperx**

### CODE REVIEW

File: src/commons/navbar-searchbar/navbar/navbar.js [hunk 1]
```diff
@@ -5,37 +5,20 @@ import { EventAggregator } from 'aurelia-event-aggregator';
...
-    ea,
-    workspaceService,
-    store,
-    appService,
-    navigationService
-  ) {
+  constructor(router, ea, store, appService, navigationService) {
     this.store = store;
     this.store.registerAction('reset', reset);
     this.subscription = this.store.state.subscribe(
```
Comment (altsop) [inline src/commons/navbar-searchbar/navbar/navbar.js:17]: Make sure that you did not forget to unsubscribe

Comment (daysleeperx) [reply src/commons/navbar-searchbar/navbar/navbar.js:17]: aren't we doing this in detached lifecycle hook? :)
```javascript
detached() {
    this.routerSubscription.unsubscribe();
    this.subscription.unsubscribe();
  }

```

Comment (altsop) [reply src/commons/navbar-searchbar/navbar/navbar.js:17]: yep, it is - haven't noticed

**Review by altsop — APPROVED** [review]
Good job

**Pull request merged**